packages/synapse/src/Synapse.ts

```diff
@@ -0,0 +1,83 @@
+
+import { type TransactionFilter, type BridgeActionParams, compressJson } from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { SynapseContract } from './contract-addresses'
+import { SYNAPSE_BRIDGE_FRAGMENTS } from './abi'
+import { CHAIN_ID_ARRAY } from './chain-ids'
+import { Token } from './Token'
+import * as tokens from './tokens'
+
+const allTokens: Token[] = Object.values(tokens)
+
+const cctpMapping: { [key: number]: string } = {
+  [1]: "0xfb2bfc368a7edfd51aa2cbec513ad50edea74e84",
+  [10]:"0xfb2bfc368a7edfd51aa2cbec513ad50edea74e84",
+  [8453]: "0xfB2Bfc368a7edfD51aa2cbEC513ad50edEa74E84",
+  [42161]: "0xfb2bfc368a7edfd51aa2cbec513ad50edea74e84",
+  [43114]:"0xfB2Bfc368a7edfD51aa2cbEC513ad50edEa74E84",
+}
+
+export const bridge = async (bridge: BridgeActionParams): Promise<TransactionFilter> => {
+  // This is the information we'll use to compose the Transaction object
+  const {
+    sourceChainId,
+    destinationChainId,
+    contractAddress,
+    tokenAddress,
+    amount,
+    recipient,
+  } = bridge
+
+  const destinationDomain = chainDomainToID(destinationChainId)
+
+  if (contractAddress && Object.values(cctpMapping).includes(contractAddress)) {
```

Comment:
Okay so basically the domain is only defined on some chains and this covers that?

Reply:
There are two different bridge contracts -- this is a case to handle when a transaction is routed through the "SynapseCCTP bridge"